blob/src/main/java/io/crate/blob/v2/BlobShard.java

```diff
@@ -56,6 +60,22 @@ public BlobShard(IndexShard indexShard, @Nullable Path globalBlobPath) {
         this.blobContainer = new BlobContainer(blobDir);
     }
 
+    public void initialize() {
+        try {
+            Files.walkFileTree(blobDir, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    blobsCount += 1;
+                    long fileSize = attrs.size();
+                    totalSize += fileSize;
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        } catch (IOException e) {
+            // we shouldn't fail because of stats
```

Comment:
@dobe Do we want to fail the blob shard because we can't read a file's properties? I think it would be nice to report it to the health monitor, but failing the recovery process and making the shard unavailable is too much IMO